cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -83,11 +83,12 @@ type writer struct {
 var _ spanstore.Writer = (*writer)(nil)
 
 // WriteSpan writes a span to the next consumer.
-func (w writer) WriteSpan(span *model.Span) error {
+func (w writer) WriteSpan(ctx context.Context, span *model.Span) error {
 	batch := model.Batch{
 		Spans:   []*model.Span{span},
 		Process: span.Process,
 	}
 	traces := jaegertranslator.ProtoBatchToInternalTraces(batch)
+	// TODO what does w.ctx represent?
```

Comment:
@pavolloffay can we get rid of `w.ctx`?

Reply:
This is because of the metrics, check the line 38, Since the context is now propagated we should modify it here.

Reply:
If I use the `ctx` arg in L93, I don't see where else `w.ctx` would be used. Should `w.ctx` be removed completely then? Or do we need to enrich `ctx` here similar to L38 and L40?

Reply:
yes the `w.ctx` can be removed and instead the context can be modified in the write method.

Reply:
ok, I made the changes, PTAL. But I think this needs to be revisited because it looks like a significant performance hit due to multiple memory allocations just to add some static labels. There should be a cheaper way of doing it, e.g. attaching a mutable structure to the Context at the start.

Reply:
Anyways this kafka receiver will go away and we will use the one from OTEL upsteream.